### PR TITLE
Update notebook environment to python 3.10

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- ~[ ] Code changes are covered by tests~
- ~[ ] Code changes have been evaluated for compatibility/integration with TrendAnalysis~
- ~[ ] New functions added to `__init__.py`~
- ~[ ] API.rst is up to date, along with other sphinx docs pages~
- [x] Example notebooks are rerun and differences in results scrutinized
- [x] Updated changelog

See #319

### CODE REVIEW

**Comment by mikofski** [comment]
😮 RdTools at the bleeding edge!

**Comment by kanderso-nrel** [comment]
> 😮 RdTools at the bleeding edge!

Not quite, they're already up to 3.11.0a7 😄 By the way, 3.11 will have some cool improvements, check it out: https://docs.python.org/3.11/whatsnew/3.11.html

**Comment by mdeceglie** [comment]
I suggest we update the notes at the top of each example to say python 3.10 instead of 3.7? Also there is a note about 3.7 and examples on the RTD index page.

Should we also re-run `degradation_and_soiling_example.ipynb`?

File: docs/sphinx/source/changelog/pending.rst [hunk 1]
```diff
@@ -14,6 +14,9 @@ Testing
 
 Requirements
 ------------
+* Upgrade the notebook environment from python 3.7 to python 3.10.
+  Several dependency versions in ``docs/notebook_requirements.txt`` are
+  updated as well. (:issue:`319`, :pull:`326`)
```
Comment (kanderso-nrel) [inline docs/sphinx/source/changelog/pending.rst:19]: Anticipating some additional dependabot PRs before the next release, I thought we might leave this as a placeholder and clean up these requirements.txt change descriptions later.

**Review by kanderso-nrel — COMMENTED** [review]
Good points, addressed.

**Review by mdeceglie — APPROVED** [review]
LGTM, thanks!

**Pull request merged**